ui/src/message_popup/mobile_prototype/message_components.jsx

```diff
@@ -0,0 +1,182 @@
+          <Paper style={messageTextStyle}>{this.props.text}</Paper>
+          
+        </div>
+        {this.props.rightIcon}
+      </div>
+    );
+  }
+});
+
+export const StudentMessage = React.createClass({
+  propTypes: {
+    text: React.PropTypes.string.isRequired,
+    student: React.PropTypes.object.isRequired,
+    onOpenStudentDialog: React.PropTypes.func.isRequired
+  },
+  
+  render(){
+    const messageStyle = _.merge({justifyContent: 'flex-start'}, styles.message);
+    const messageTextStyle = _.merge({backgroundColor: '#f1c889'}, styles.messageText);
+    
+    return (
+      <div>
+        <Message 
+          type="student"
+          text={this.props.text}
+          label={this.props.student.name}
+          messageStyle={messageStyle}
+          messageTextStyle={messageTextStyle}
+          leftIcon={
+            <IconButton 
+              onTouchTap={this.props.onOpenStudentDialog} 
+              style={styles.messageIconButton} 
+              iconStyle={styles.messageIcon}>
+              <FaceIcon/>
+            </IconButton>
+          }
+          />
+      </div>
+    );
+  }
+});
+
+export const UserMessage = React.createClass({
+  propTypes: {
+    text: React.PropTypes.string.isRequired,
+    label: React.PropTypes.string
+  },
+
+  
+  render(){
+    const messageStyle = _.merge({justifyContent: 'flex-end'}, styles.message);
+    const messageTextStyle = _.merge({backgroundColor: '#e6f9ff'}, styles.messageText);
+    
+    return (
+      <div>
+        <Message 
+          type="user"
+          text={this.props.text}
+          label={this.props.label === undefined ? 'You' : this.props.label}
+          messageStyle={messageStyle}
+          messageTextStyle={messageTextStyle}
+          rightIcon={
+            <FaceIcon  style={{marginTop: 8, ...styles.messageIcon}}/> 
+          }
+          />
+      </div>
+    );
+  }
+});
+
+export const InfoMessage = React.createClass({
+  propTypes: {
+    text: React.PropTypes.string.isRequired,
+    onOpenInfoDialog: React.PropTypes.func.isRequired
+  },
+  
+  render(){
+    const messageStyle = _.merge({justifyContent: 'flex-start'}, styles.message);
+    const messageTextStyle = styles.messageText;
+    
+    return (
+      <div>
+        <Message 
+          type="info"
+          text={this.props.text}
+          messageStyle={messageStyle}
+          messageTextStyle={messageTextStyle}
+          leftIcon={
+            <IconButton 
+              onTouchTap={this.props.onOpenInfoDialog}
+              style={styles.messageIconButton}
+              iconStyle={styles.messageIcon}>
+              <InfoOutlineIcon/>
+            </IconButton>
+          }
+          />
+      </div>
+    );
+  }
+});
+
+const styles = {
+  message: {
+    display: 'flex',
+    padding: 5
+  },
+  messageIconButton: {
+    margin: 0,
+    padding: 0,
+    flexGrow: 0,
+    flexShrink: 0,
+    flexBasis: 30
+  },
+  messageIcon: {
+    width: 30,
+    height: 30,
```

Comment:
I tried changing the fixed width and height by relying on setting just the flexBasis and using margins to position it but the IconButton wave effect looks really awkward and was off-centered.